test/generator.ml

```diff
@@ -0,0 +1,59 @@
+open Messages
+
+module type MTest = sig
+  open QCheck2.Gen
+  open Pollinate
+  val peer_gen : Peer.t t
+
+  val address_gen : Address.t t
+
+  val peer_print : Peer.t -> string
+
+  val request_gen : Messages.request t
+
+  val peer_status_gen : Peer.status t
+
+  val sockaddress_gen : Unix.sockaddr t
+end
+
+module Test : MTest = struct
+  open QCheck2.Gen
+  open Pollinate
+  let address_gen =
+    pair (pure "127.0.0.1") int >|= fun (address, port) ->
+    Address.{ address; port }
+
+  let peer_gen =
+    let open Peer in
+    address_gen >|= fun address ->
+    { address; status = Alive; neighbors = Base.Hashtbl.Poly.create () }
+
+  let peer_print peer =
```

Comment:
We can just put this in the peer module so consumers can use it too if they wish

Reply:
Of couse, great idea. Also added the `status`.